sahha-android/src/main/java/sdk/sahha/android/domain/use_case/metadata/AddSahhaDataLogMetadata.kt

```diff
@@ -0,0 +1,37 @@
+package sdk.sahha.android.domain.use_case.metadata
+
+import sdk.sahha.android.common.SahhaTimeManager
+import sdk.sahha.android.domain.model.data_log.SahhaDataLog
+import sdk.sahha.android.domain.model.data_log.SahhaMetadata
+import sdk.sahha.android.domain.repository.BatchedDataRepo
+import javax.inject.Inject
+
+internal class AddSahhaDataLogMetadata @Inject constructor(
+    private val timeManager: SahhaTimeManager,
+    private val repo: BatchedDataRepo,
+) {
+    suspend operator fun invoke(
+        batchedData: List<SahhaDataLog>,
+        postDateTime: String = timeManager.nowInISO()
+    ): List<SahhaDataLog> {
+        val modifiedData = mutableListOf<SahhaDataLog>()
+
+        batchedData.forEach { dataLog ->
+            val metadata = dataLog.metadata
+            val postDateTimes = metadata?.postDateTime?.toMutableList()
+
+            modifiedData += postDateTimes?.let { times ->
+                appendPostDateTime(dataLog.copy(metadata = SahhaMetadata(times)), postDateTime)
+            } ?: appendPostDateTime(dataLog, postDateTime)
+        }
+
+        repo.saveBatchedData(modifiedData)
+        return modifiedData
+    }
+
+    private fun appendPostDateTime(dataLog: SahhaDataLog, postDateTime: String): SahhaDataLog {
```

Comment:
The functionality for creating and appending metadata should be generic. If you are reusing code (across sleep, steps, etc.), it should be a method external to the calling class. Please move this to one place and call from the child objects.

Reply:
yes I initially thought that too but had trouble coming up with a solution and didn't know how long it might take. Managed to do it this morning bouncing some ideas with mr GPT